version update to 1.12

Aaronstar-McClure opened the pull request:
Trying to update to 1.12. Just making the pr for now, will try adding the changes I have so far when I get home. 
Edit: forgot to mention, issue #6

Aaronstar-McClure committed:
initial commit so I can start a pr

Aaronstar-McClure committed:
update gradle

updated gradle and scripts. tried updating old files but would not work. going to re add important lines to build.gradle later, but this is necessary to even get the workspace setup

Aaronstar-McClure committed:
add bspkrscore as dep in gradle. doesnt work, but it is there...

Aaronstar-McClure commented:
updated gradle so i could initialize the dev environment

Aaronstar-McClure committed:
sorted imports

sorts the imports, removes unused ones, and potentially corrects broken ones where methods were moved

Aaronstar-McClure committed:
clean floatingruins.java part 1

todo: fix config generation at the bottom, need to convert to modname:itemname.

verify: ambiguous corrected import; minecraftforge.fml.common.registry.GameData no longer exists. auto corrected to net.minecraftforge.registries.GameData this is an internal registry and should not be used. net.minecraftforge.fml.common.registry.GameRegistry is the closest to original version and may have necessary functions to correct.

Aaronstar-McClure committed:
clean commandfrgen.java

verify correct change to compareTo

Aaronstar-McClure committed:
clean worldGenFloatingIsland.java pass 1

again, only thing left is the registry problem

Aaronstar-McClure committed:
clean WorldGenFloatingIslandRuin.java pass 1

todo: figure out biome checker and loot table system

Aaronstar-McClure committed:
clean floatingRuinsWorldGenerator.java

simple

Aaronstar-McClure committed:
clean FloatingRuinsTicker

Aaronstar-McClure committed:
clean modGuiFactoryHandler.java and delete GuiFRConfig.java

see comments

Aaronstar-McClure committed:
sort imports again, just like it clean

Aaronstar-McClure committed:
oops, almost forgot this one...

Aaronstar-McClure commented:
well, that is the first pass of fixes. mostly easy, some a bit complicated.
todo: correct the many and varied instances of GameData.getItemRegistry().getNameForObject 
this seems to rely on public item id's, which is no longer a thing...
todo: figure out the loot system as chestGenHooks is no longer a thing....

Aaronstar-McClure commented:
@bspkrs if you are interested in looking at this and giving me pointers that would defiantly help. this is my first real dive into modding, so i really would like to figure this out

Aaronstar-McClure committed:
added version info

MichaelHillcox commented:
Looks good so far, I can do a proper review of it later on. I'll talk to Bspkrs sometime soon about getting this repo moved to the org.

Aaronstar-McClure commented:
Cool, currently I'm looking into if I can pull the core dep from github rather then the bspk.rs repo. That might make it a bit eaiser and have fewer steps in the future...

Aaronstar-McClure commented:
Here's something that might help
Do you know the groupName:moduleName for bspkrsCore? I can't seem to get Gradle to resolve the dependency from GitHub

MichaelHillcox commented:
You can use cursemaven.com to pull it into the project :)

Aaronstar-McClure committed:
this will not resolve...

Aaronstar-McClure commented:
i cant seem to get the dependency to resolve... any thoughts?

Aaronstar-McClure committed:
think I fixed the trouble with items. got to test it once I can compile

Aaronstar-McClure committed:
finally got the dep import figured out

turns out there was an extra step not in the instructions where i had to add a gradle plugin...

Aaronstar-McClure commented:
Turns out the maven link did not want to work, so I had to use a gradle plugin instead to grab from curse. It should work now...
